packages/database/src/core/Repo.ts

```diff
@@ -466,15 +469,50 @@ export function repoGetValue(repo: Repo, query: QueryContext): Promise<Node> {
   }
   return repo.server_.get(query).then(
     payload => {
-      const node = nodeFromJSON(payload as string).withIndex(
+      const node = nodeFromJSON(payload).withIndex(
         query._queryParams.getIndex()
       );
-      const events = syncTreeApplyServerOverwrite(
-        repo.serverSyncTree_,
-        query._path,
-        node
-      );
-      eventQueueRaiseEventsAtPath(repo.eventQueue_, query._path, events);
+      // if this is not a filtered query, then overwrite at path
+      if (query._queryParams.loadsAllData()) {
+        const events = syncTreeApplyServerOverwrite(
+          repo.serverSyncTree_,
+          query._path,
+          node
+        );
+        eventQueueRaiseEventsAtPath(repo.eventQueue_, query._path, events);
```

Comment:
Looking at this again, I suspect we may need to add a call to `repoRerunTranactions` call here. 

We do so in the data update handling code for a listener: https://github.com/firebase/firebase-js-sdk/blob/master/packages/database/src/core/Repo.ts#L400-L406.

Reply:
(same thing for L504, after `applyTaggedQueryMerge`)

Reply:
I don't think we need to do this as `repoOnDataUpdate` does this for the listeners